e2e-tests/ukef/cypress/e2e/journeys/utilisation-reports/utilisation-report-submission.spec.js

```diff
@@ -0,0 +1,235 @@
+   * values from the report as defined in february2023ExpectedValues
+   * @param {number} utilisationTableRowIndex - The index of the row of utilisation table
+   * @param {object} expectedParsedValuesForReportRow - The expected values to be found in the row
+   */
+  const assertUtilisationTableRowContainsExpectedValues = (utilisationTableRowIndex, expectedParsedValuesForReportRow) => {
+    tfmPages.utilisationReportPage.utilisationTab.table.rowByIndex(utilisationTableRowIndex).within(() => {
+      cy.assertText(tfmPages.utilisationReportPage.utilisationTab.table.facilityId(), expectedParsedValuesForReportRow.facilityId);
+      cy.assertText(tfmPages.utilisationReportPage.utilisationTab.table.exporter(), expectedParsedValuesForReportRow.exporter);
+      cy.assertText(tfmPages.utilisationReportPage.utilisationTab.table.feesPayable(), expectedParsedValuesForReportRow.reportedFees);
+      cy.assertText(tfmPages.utilisationReportPage.utilisationTab.table.feesAccrued(), expectedParsedValuesForReportRow.feesAccrued);
+      cy.assertText(tfmPages.utilisationReportPage.utilisationTab.table.utilisation(), expectedParsedValuesForReportRow.utilisation);
+    });
+  };
+
+  const testFebruary2023ReportUploadWithFileType = (filetype) => {
+    const filename = filetype === 'csv' ? 'valid-utilisation-report-February_2023_monthly.csv' : 'valid-utilisation-report-February_2023_monthly.xlsx';
+
+    //---------------------------------------------------------------------------------
+    // Portal payment report officer submits utilisation report
+    //---------------------------------------------------------------------------------
+    cy.login(BANK1_PAYMENT_REPORT_OFFICER1);
+    cy.visit(relative('/utilisation-report-upload'));
+
+    portalPages.utilisationReportUpload.utilisationReportFileInput().attachFile(filename);
+    cy.clickContinueButton();
+    portalPages.confirmAndSend.confirmAndSendButton().click();
+
+    //---------------------------------------------------------------------------------
+    // PDC_RECONCILE user can login to TFM and view the data from the submitted report
+    //---------------------------------------------------------------------------------
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+
+    cy.forceVisit(TFM_URL);
+
+    cy.tfmLogin(PDC_RECONCILE);
+    cy.url().should('eq', `${TFM_URL}/utilisation-reports`);
+
+    tfmPages.utilisationReportsSummaryPage.reportLink(BANK1_PAYMENT_REPORT_OFFICER1.bank.id, '2023-03').click();
+
+    /**
+     * Each line of the report will have an entry in the premium payments table,
+     * but they will not be in the same order so first we find the index within
+     * the premium payments tab table for each of the report rows.
+     */
+    tfmPages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().each(($row, index) => {
+      cy.wrap($row)
+        .find('[data-cy="exporter"]')
+        .eq(0)
```

Comment:
Are we able to use selectors and `rowIndex` RE a previous comment?

Reply:
Sorry, I'm not sure what you mean? I didn't put this in pages since it is quite complicated and I'm not sure I see this being relevant to many other tests since in most of our tests we want to get the row by the data-cy which uses the feeRecordId since in most tests that is the most usual way to get it, otherwise we would have to do essentially this in all those tests and that would be in a lot more tests. Would be nice if we could have two different data-cy on one element so we can get them by different properties in different tests but I haven't seen a way to do that.

Reply:
The problem is most of our e2e-tests get the rows by feeRecordId and that is generally much more useful than by exporter since exporter is not guaranteed to be unique, so feerecordId is used in the data-cy. Here I want to get by exporter because I have picked exporter name that are unique to allow for this testing.

It would be nice if we could have multiple data-cy per element but as far as I can see that is not possible and I think this is the simplest I can manage this.

Reply:
I think it's possible to have multiple data-cy values separated by a space and a different cypress selector - trying this out in another PR like this:

```
"data-cy": "deal__exporter--index--1 deal__exporter--abc12345678"
```

So we can get a row by index, or deal ID.  E.g by ID, the "*" after `data-cy` is for if `data-cy` contains the value

```js
exporter: (id) => cy.get(`[data-cy*="deal__exporter--${id}"]`)
```

Reply:
Ooh interesting, I'll give this a go to see if it can help me simplify!

Reply:
This is now simplified greatly, as it removed the need to loop through the rows and store the indexes.